### PR TITLE
Spring security

### PR DESCRIPTION
**Pull request by alisonreynoso**
## Summary
- Completed spring security setup
- Added a test profile to run test without login
## Trello Card
https://trello.com/c/mJ94EilH/23-integrar-spring-security

### CODE REVIEW

File: src/main/java/wolox/training/security/PasswordEncoder.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+@Configuration
+public class PasswordEncoder {
+
+    @Bean
+    public org.springframework.security.crypto.password.PasswordEncoder encoder() {
+        return new BCryptPasswordEncoder();
+    }
+    public String encode(String string){
```
Comment (julianvasquez-wolox) [inline src/main/java/wolox/training/security/PasswordEncoder.java]: Is this method really used? when you create a java Bean and it's injected using Autowired then you get the object returned in the @ Bean method ( BCryptPasswordEncoder for this case ) so any method will be defined by the object and not by de config class ( unless you override them )

File: src/main/java/wolox/training/security/CustomUserDetailesService.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import wolox.training.models.User;
+import wolox.training.repositories.UserRepository;
+
+@Service
+@Profile("!test")
+public class CustomUserDetailesService implements UserDetailsService {
```
Comment (julianvasquez-wolox) [inline src/main/java/wolox/training/security/CustomUserDetailesService.java]: Check the class name 😅

File: src/main/java/wolox/training/security/PasswordEncoder.java (hunk 1 above)
Comment (alisonreynoso) [reply src/main/java/wolox/training/security/PasswordEncoder.java]: I used in order to mock it in tests,but I could find another way to mock it

Comment (alisonreynoso) [reply src/main/java/wolox/training/security/PasswordEncoder.java]: Oh, now I get it ! And there was no need for the method haha! Thanks

**Review by julianvasquez-wolox — APPROVED** [review]
great! 🥳

File: src/test/java/wolox/training/models/BookEntityTest.java [hunk 3]
```diff
@@ -84,17 +84,17 @@ public void whenBookItIsSaved_thenYouCanGetItBackWithId(){
...
+        Book book2 = bookRepository.findById(book1.getId()).get();
+        assertTrue(book1.getId() == book2.getId());
+        assertTrue(book1.getAuthor() == book2.getAuthor());
+        assertTrue(book1.getImage() == book2.getImage());
+        assertTrue(book1.getYear() == book2.getYear());
+        assertTrue(book1.getPages() == book2.getPages());
+        assertTrue(book1.getIsbn() == book2.getIsbn());
+        assertTrue(book1.getPublisher() == book2.getPublisher());
+        assertTrue(book1.getSubtitle() == book2.getSubtitle());
+        assertTrue(book1.getTitle() == book2.getTitle());
```
Comment (Jimenarosello) [inline src/test/java/wolox/training/models/BookEntityTest.java]: you can use assertEquals :)

**Review by Jimenarosello — APPROVED** [review]
Nice job! 😄

File: src/test/java/wolox/training/models/BookEntityTest.java (hunk 3 above)
Comment (alisonreynoso) [reply src/test/java/wolox/training/models/BookEntityTest.java]: Done :D

**Pull request merged**